### DISCUSSION TITLE
Trying to push to a private repo and asking for username and password

### DISCUSSION THREAD

**Discussion by chris-nt** [community/community#133367] [New to GitHub] [answered]
Hello, I am a new user to github. So I am trying to push to a private repository my project and I am getting prompted to enter my username and password. As far as I know i shouldn't be prompted with that. I know that i have to use an SSH key, that i have already created it and put it in my account but yet it stills asks me for username and password. Any ideas?
Thanks in advance!

**Comment by davevad93** [comment, marked as answer]
Hi @xfosnt ,

When performing git operations with `SSH` such as `git clone` or `git push` you will be prompted to enter just the password you set when generating the `ssh key`, see the documentation for more information:

- [Generating a new SSH key and adding it to the ssh-agent](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)

On the other hand, when performing git operations with `HTTPS` it prompts you to enter your GitHub credentials every time you pull or push a repo and the password has to be a `Personal Access Token` instead of your account password, for more info see the docs:

- [Why is Git always asking for my password?](https://docs.github.com/en/get-started/getting-started-with-git/why-is-git-always-asking-for-my-password)
-  [Managing your personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens)

You can choose how to clone a repo by selecting `HTTPS` or `SSH`. If you clone a repo with `HTTPS` when you push the changes it will be done with `HTTPS` because the remote origin reference has been set to `https://github.com/username/repo.git` instead of `git@github.com/username/repo.git`.

![0](https://github.com/user-attachments/assets/f87de7f0-e4da-4359-bb4b-e567753e3638)

**Comment by chris-nt** [comment]
Oh okay thank you a lot for your help!
